src/Orchard/ContentManagement/DefaultHqlQuery.cs

```diff
@@ -207,12 +207,18 @@ public class DefaultHqlQuery : IHqlQuery {
 
         public int Count() {
             ApplyHqlVersionOptionsRestrictions(_versionOptions);
-            var hql = ToHql(true);
-            hql = "select count(Id) from Orchard.ContentManagement.Records.ContentItemVersionRecord where Id in ( " + hql + " )";
-            return Convert.ToInt32(_session.CreateQuery(hql)
-                           .SetCacheable(true)
-                           .UniqueResult())
-                ;
+            var sql = ToSql(true);
+            sql = "SELECT count(*) as totalCount from (" + sql + ") t";
```

Comment:
I've just checked now this code works properly with MySql and Postgres executing a projection query with pagination.